### ISSUE TITLE
[BUG/ISSUE] Possible coding error in function HETClNO3_HCl in gckpp_HetRates.F90

### ISSUE THREAD

**Issue by eamarais** [geoschem/geos-chem#750]
### Description of the problem
I suspect there's a coding error in the function HETClNO3_HCl in gckpp_HetRates.F90 that is causing heterogeneous reaction rates to only be calculated for aerosol type N=13. I've included @sdeastham, as he'll be able to confirm/refute my finding and also comment on whether my minor change to the code is appropriate. 

### Description of troubleshooting performed
I am running perturbation simulations with additional emissions that should cause loss of ozone in the stratosphere via the ClONO2+HCl-->HNO3+Cl2 reaction, but this led to no change using the code as is. Only after I edited the code, did I see the decline in ozone I'd expect. 

### GEOS-Chem version
Version 12.9.3, but it is also present in version 13.0.2.

### Description of modifications
I have changed the code from:
         IF (STRATBOX.and.(N.eq.13)) THEN
            ! Calculate for stratospheric liquid aerosol
            ! Note that XSTKCF is actually a premultiplying
            ! factor in this case, including c-bar
            ADJUSTEDRATE = XAREA(N) * XSTKCF
         ELSE IF (XSTKCF .GT. 0.0e+0_fp) THEN
            ! Reaction rate for surface of aerosol
            ADJUSTEDRATE=ARSL1K(XAREA(N),XRADI(N),XDENA,XSTKCF,XTEMP, &
                                                        XSQM)
         ENDIF

To:
         IF (XSTKCF .GT. 0.0e+0_fp) THEN
            IF (N.eq.13) THEN
               ! Calculate for stratospheric liquid aerosol
               ! Note that XSTKCF is actually a premultiplying
               ! factor in this case, including c-bar
               ADJUSTEDRATE = XAREA(N) * XSTKCF
            ELSE
               ! Reaction rate for surface of aerosol
               ADJUSTEDRATE=ARSL1K(XAREA(N),XRADI(N),XDENA,XSTKCF,XTEMP, &
                                                            XSQM)
           ENDIF
         ENDIF

I've also attached a screenshot of my Git GUI window that shows the line changes:

**Comment by eamarais** [comment]
Forgot to attach the screenshot: 
![Screenshot-het-rxn-hetclno3_hcl-update](https://user-images.githubusercontent.com/55648553/121884158-7b7ba580-cd0a-11eb-896c-8cf7f7f3cd11.png)

**Comment by sdeastham** [comment]
I think I'm missing something - why does the new code change the behavior? If the box is stratospheric and `N == 14` (i.e. ice/NAT PSC), then `XSTKCF` = 0.2 or 0.3 and we should follow the second branch of the logic (`ELSE IF (XSTKCF .GT. 0.0e+0_fp) THEN...`. The only functional change I see is that the `STRATBOX` requirement is removed, but that should effectively be enforced by the clause which only defines `XSTKCF` as greater than zero when `STRATBOX` is true. What am I missing?

At some point we should also look into why the chemistry is only calculated in the stratosphere, but that's a separate issue!

**Comment by eamarais** [comment]
Ah, @sdeastham you're right. Not sure why at the time I thought it was skipping the calculation for the other aerosol types. I guess this can be closed and maybe removed from the issues list.

**Comment by yantosca** [comment]
Thanks @sdeastham and @eamarais.  I have been working on updating the het chem rates for both clarity & efficiency.  I've discovered that many of the terms in the various routines only need to be computed once for each (I,J,L) grid box instead of for each subroutine call.  Also, I've removed much of the looping & branching, as to loop over 14 aerosol types but only to do something with 2 or 3 of them is very wasteful of CPU cycles.  I'm almost done with this but also need to balance this work with adding new science into GEOS-Chem, so it may take a bit longer as I'm juggling tasks.
